server/database.py

```diff
@@ -10,43 +12,85 @@ def connect_to_db(config):
 def store_game(db_connection, game_data):
     cursor = db_connection.cursor()
 
-    columns_str = ("Time, Task, b, W, G, bm, br, pn, pp, te, ie, p, " +
-                   "TimeAndLocation, InitialInfo, SearchWords, finalScore, standard_info")
-    values_str =  n_values_str(17)
-    query_str = "INSERT INTO ibmdb(" + columns_str + ") values(" + values_str + ")"
+    game_id = str(uuid.uuid4())
+    game_query= "INSERT INTO game(id, final_score, start_time, total_time) VALUES(" + n_values_str(4) + ")"
+
+
+    game_values = (game_id,
+                   game_data['finalScore'],
+                   dt.datetime.fromtimestamp(int(game_data['startTime']) / 1e3),
+                   game_data['time'])
+    cursor.execute(game_query, game_values)
+
+    block_fronts = {}
+    for block in game_data['initialInfo']:
+        block_id = block['id']
+        front_color = block['color']
+        front_letter = block['letter']
+        block_fronts[block_id] = (front_color, front_letter)
+
+        block_query = "INSERT INTO block(id, game_id, front_color, back_color, front_letter, back_letter) VALUES( " + n_values_str(6) + ")"
+        block_values = (block_id,
+                        game_id,
+                        front_color,
+                        block['flipColor'],
+                        front_letter,
+                        block['flipLetter'])
+        cursor.execute(block_query, block_values)
+
+    for action in game_data['actions']:
+        action_type = action['type']
```

Comment:
Similarly, if we want to go a little wild, we could split each action type into its own function.